python/ray/_private/runtime_env/nsight.py

```diff
@@ -0,0 +1,132 @@
+    try:
+        nsight_cmd = nsight_cmd + ["python", "-c", '""']
+        result = subprocess.run(
+            nsight_cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        )
+        if result.returncode == 0:
+            subprocess.run(["rm", "empty.nsight-rep"])
+            return True, None
+        else:
+            subprocess.run(["rm", "empty.nsight-rep"])
+            error_msg = (
+                result.stderr.strip()
+                if result.stderr.strip() != ""
+                else result.stdout.strip()
+            )
+            return False, error_msg
+    except FileNotFoundError:
+        return False, ("nsight is not installed")
+
+
+def parse_nsight_config(nsight_config: Dict[str, str]) -> List[str]:
+    """
+    Function to convert dictionary of nsight options into
+    nsight command line
+
+    The function returns:
+    - List[str]: nsys profile cmd line split into list of str
+    """
+    nsight_cmd = ["nsys", "profile"]
+    for option, option_val in nsight_config.items():
+        if option[:2] == "--":
+            nsight_cmd.append(f"{option}={option_val}")
+        elif option[0] == "-":
+            nsight_cmd += [option, option_val]
+    return nsight_cmd
+
+
+class NsightPlugin(RuntimeEnvPlugin):
+    name = "nsight"
+
+    def __init__(self, resources_dir: str):
+        self.nsight_cmd = []
+
+        # replace this with better way to get logs dir
+        session_dir, runtime_dir = os.path.split(resources_dir)
+        self._logs_dir = os.path.join(session_dir + "/logs/nsight")
+        try_to_create_directory(self._logs_dir)
+
+    async def create(
+        self,
+        uri: Optional[str],
+        runtime_env: "RuntimeEnv",  # noqa: F821
+        context: RuntimeEnvContext,
+        logger: logging.Logger = default_logger,
+    ) -> int:
+        nsight_config = runtime_env.get("nsight")
+        if not nsight_config:
+            return 0
+
+        if isinstance(nsight_config, str):
+            if nsight_config == "default":
+                nsight_config = NSIGHT_DEFAULT_CONFIG
+            else:
+                raise RuntimeError(
+                    f"Unsupported nsight config: {nsight_config}. "
+                    "The supported config is 'default' or "
+                    "Dictionary of nsight options"
+                )
+
+        is_valid_nsight_cmd, error_msg = check_nsight_script(nsight_config)
+        if not is_valid_nsight_cmd:
+            logger.warning(error_msg)
+            raise RuntimeError(
+                "nsight profile failed to run with the following "
+                f"error message:\n {error_msg}"
+            )
+        # add set output path to logs dir
+        nsight_config["-o"] = f"{self._logs_dir}/" + nsight_config.get(
+            "-o", NSIGHT_DEFAULT_CONFIG["-o"]
+        )
+
+        self.nsight_cmd = parse_nsight_config(nsight_config)
+        return 0
```

Comment:
Q: what's the plan for GC policy now?

Reply:
Also, cc @architkulkarni how do we currently GC runtime resources? Which directory should we use to utilize the default GC features? (Also, should `create` return the size of artifact to utilize this feature?)

Reply:
`create` in runtime_env is used if we requires installing some packages (like pip, or working_dir)
So I set it to 0 as we don't install the dependency for user, only modifying the RuntimeContext

Reply:
I see. I guess the file generated by a profiler is a separate question. do you plan to tackle it in the followup prs?